src/libs/antares/study/parts/short-term-storage/series.cpp

```diff
@@ -180,4 +180,59 @@ bool Series::validateLowerRuleCurve() const
     return checkVectBetweenZeroOne(maxInjectionModulation, "lower rule curve");
 }
 
+bool Series::validateInitialLevelSimplex(bool simplexIsWeek, std::optional<double> level,
+        unsigned int cycleDuration, unsigned int startHour, unsigned int endHour) const
+{
+    if (startHour > endHour)
+        endHour += HOURS_PER_YEAR;
+
+    unsigned int simulationHour = simplexIsWeek ? 168 : 24;
+
+    if (level.has_value())
+        return checkLevelValue(level.value(), cycleDuration, startHour, endHour) &&
+                checkLevelValue(level.value(), simulationHour, startHour, endHour);
+    else
+        return checkLevelInterval(cycleDuration, startHour, endHour) &&
+                checkLevelInterval(simulationHour, startHour, endHour);
+}
+
+bool Series::checkLevelValue(double level, unsigned int cycleDuration, unsigned int startHour, unsigned int endHour) const
```

Comment:
I'm not sure that this function **checkLevelValue(...)** does what we want : 
- Imagine we have a simulation 2 weeks long (from h = 0 to hour 2 * 168 - 1= 335 hours).
- Imagine it is a weekly optimization (over 168 hours blocks)
- Imagine we have a cycle of 100 hours.

What the code here does is : make a check on initial level at hours 0, 100, 168, 200, 300.
What we want is : make a check at hours : 0, 100, 168 and 268.

Reply:
For every week (separably) 
- Take the intersection of [Lower bound[h], Upper Bound[h]] for hours h=0, 99
- Check it is not empty
- Save it for the modeler structure `std::vector<BoundsForInitialLevel> bounds`
- The modeler fetches `bounds[weekIndex]` for every week, see function `SIM_RenseignementProblemeHebdo`